.github/workflows/ci.yml

```diff
@@ -3,89 +3,110 @@ name: CI
 on:
```

Comment:
should probably add the cancel workflow action to cancel jobs on commits that have been superceeded in order to save build resources. I set it up in fw so can take a look there. could be a follow up thing

Reply:
done